### DISCUSSION TITLE
RustDesk refuses to use my OpenVPN network adapter but uses my Smartphone Tethering one

### DISCUSSION THREAD

**Discussion by SpanishHearts** [rustdesk/rustdesk#7707] [General]
### Bug Description

Well the headline says it all. Everything is properly setup, and I have awesome access from everywhere to everything but for some reason on Windows 11, RustDesk refuses to use my OpenVPN connection and uses the one I share from my smartphone (hence using this IP as trying to connect to my self-hosted server)

OpenVPN is working on my PC for EVERY SINGLE OTHER application, but for some reason, RustDesk doesn't use it.

How can I Force RustDesk to use a particular Network Adapter for trying to connect to my server? 

### How to Reproduce

Setup a hotspot via your smartphone , either via USB or WiFI tethering (both doesn't work)
Setup a home server and try to limit the access only to your OpenVPN IP
You won't be able to connect
Now open the access to the IP of the tethered connection - you will be able to connect - hence RustDesk is not using the VPN

### Expected Behavior

Nothing

### Operating system(s) on local side and remote side

Windows 11

### RustDesk Version(s) on local side and remote side

1.2.3-1.2.3

### Screenshots

![image](https://github.com/rustdesk/rustdesk/assets/87440293/2660b0e4-354e-4e58-a894-de9f386d3590)

**Comment by DMSnz** [comment]
We use OpenVPN and I do not have any such issues, self host Pro, clients everywhere and in fact ONLY able to connect via VPN (so not public open RustDesk server).

Is your RustDesk instance on a different subnet, that needs a route added in your OpenVPN config?
Do you have any routes, that then forces "local" traffic over the VPN, rather than via public internet?

I have run it like this since day one, so its not a RustDesk issue I am reasonably certain.
Windows 10/11 clients galore no issue.

**Reply by SpanishHearts** [reply]
If it's not RustDesk issue, how so only RustDesk has this issue on the computer?

It's not on a different subnet, and also it has no issue when connecting from OpenVPN from my phone with the same VPN settings. It has issue only on the PC and on the same PC no other app or local network has this issue.

I replicated the same issue on 3 other laptops running Windows 11.

**Reply by DMSnz** [reply]
Assuming you use ovpn config files, is it possible the VPN config between computers and your phone are different somehow?
Typically the OpenVPN server should spit out configs with the same conditions and conditions set to your requirements, but not to say one was created earlier before a change, or edited after the fact maybe?
Or is it that a firewall on Windows is treating VPN as an interface that needs its own rules, and possibly the phone hotspot network is treated as another where such traffic is allowed?

**Comment by SpanishHearts** [comment]
It's physically and mathematically impossible, because I'm using a USB-C flash drive with only one configuration file. 

Again - I have no issues with any other app that requires VPN, every single other app detects and use the proper network adapter - the OpenVPN one. 

Only RustDesk demands to connect to the tethering adapter.

**Comment by DMSnz** [comment]
what happens if you trace route your rustdesk server from both laptop vs OpenVPN and then phone tether VPN?

**Comment by SpanishHearts** [comment]
Nothing, it shows that RustDesk is trying to reach the RustDesk server via the phone tether. 

If I OpenVPN on the phone first and then tether, nothing happens. It's still using the phone's original IP (OpenVPN on the phone does not affect the hotspot connections - android limitation)

**Reply by DMSnz** [reply]
So with no phone connected to laptop
Laptop connected with OpenVPN
It still trace routes via phone tether?
Disconnect everything, restart (not shutdown) connect VPN without phone on laptop only and then trace route your RustDesk server.

**Comment by SpanishHearts** [comment]
No, no no. With no phone connected there is no internet. If I connect via my wifi and test with OpenVPN it works (because it still prefers the wifi, which is a local network)

**Comment by DMSnz** [comment]
Oh so the issue is it does not work when tethered/hotspot?
To clarify, OpenVPN on the computer, but when tethered/hotspot as internet it does not work?
But if you OpenVPN client on the phone, then tether/hotspot it does work?

**Comment by SpanishHearts** [comment]
Yes, exactly. Also on the computer, everything else works as expected with Hotspot and OpenVPN, except RustDesk.

Btw, if I open my ports and allow the hotspot IP to pass to my firewall (the phone IP), it starts working (hence RustDesk adopting by default the hotspot IP and not the OpenVPN one)

**Comment by DMSnz** [comment]
I by chance have OpenVPN on my phone, so will test this scenario and post back if I get the same issue or not.

**Comment by SpanishHearts** [comment]
OpenVPN on the phone works perfectly with RustDesk via mobile network. OpenVPN doesn't work on Windows 11 with RustDesk (but works with everything else)

If I check my mobile hotspot IP and allow it through my firewall, magically RustDesk on Windows 11 starts working. If I block the phone hotspot IP - stops working only with RustDesk (ONLY ON THE WINDOWS it continues working via OpenVPN on android)

**Comment by DMSnz** [comment]
Just trying to hijack someones laptop to test,
By any chance do you use Eset Security?
I have found lately I can not for the life of me connect to Rustdesk clients without first disabling Eset Firewall and thats not even considering VPN, just in general.

**Comment by SpanishHearts** [comment]
No, I have no antiviruses on my computers since 1998 (knock on wood) 

I have tested on my wife's laptop (also windows 11) and on my neighbour. Same exact effect. No matter how I set it up, RustDesk always choose the hotspot network adapter, instead of the OpenVPN adapter. I tried to block it via firewall, to set prioritization on the OpenVPN one instead of the Hotspot one - no luck.

It's definitely something with RustDesk, because otherwise I can connect to every single local app (and I'm running a lot of A.I. / management stuff + NAS)

**Comment by DMSnz** [comment]
tested just now
a Windows 11 laptop, with OpenVPN on the laptop and my phone as hotspot (with phone wifi disabled so not on our network).
This works no issue at all, RustDesk says ready on client and I can connect remotely from my laptop to it.

Not sure if any use, but the OpenVPN link on laptop does not have a gateway set, only the wireless hotspot link shows a gateway
My test laptop using OVPN + Hotspot has client version 1.2.3-1, my laptop which I connected to it from has 1.2.4
Server Pro is 1.2.0 for now about to be updated.

Seemingly working fine my end.

**Reply by SpanishHearts** [reply]
That gave me a glimpse of hope, because I had enabled the gateway option in the OpenVPN config. I edited it out but sadly it's not working. RustDesk still demands to use the tethered hotspot IP...

According to your setup - can you share your Firewall settings (you can blur out the IP's). I want it just for port reference

**Comment by DMSnz** [comment]
I have no firewall opening ports for RustDesk as such...
We use a proxy to wrap https on everything as needed and RustDesk is no exception to this.
Our OpenVPN server runs in its own IP range/subnet so all we had to ensure, was OpenVPN could "see" the routes to our main network where RustDesk resides.
Due RustDesk being docker also, its not hard wired with its own IP, it gets access via the proxy.

**Reply by DMSnz** [reply]
I could possibly share OpenVPN config settings we use, would prefer to do that via a more private channel though?
